src/main/python/pybuilder/plugins/python/core_plugin.py

```diff
@@ -67,8 +67,9 @@ def list_scripts():
         if not os.path.exists(scripts_dir):
             return
         for script in os.listdir(scripts_dir):
-            if os.path.isfile(os.path.join(scripts_dir, script)):
-                yield script
+            if HIDDEN_FILE_NAME_PATTERN.match(script):
```

Comment:
What about sub-directories? Looks like that we just need to add condition to existing: 
```
            if os.path.isfile(os.path.join(scripts_dir, script)) and not HIDDEN_FILE_NAME_PATTERN.match(script):
                yield script

```